yt/yt/server/http_proxy/solomon_proxy.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "public.h"
+#include "component_discovery.h"
+
+#include <yt/yt/library/profiling/solomon/proxy.h>
+
+namespace NYT::NHttpProxy {
+
+////////////////////////////////////////////////////////////////////////////////
+
+//! Works for all components represented in the `EYTComponentType` enum.
+class TYTComponentEndpointProvider
```

Comment:
Currently, the name does not indicate that this class provides _profiling endpoints for sensor fetching_. If this class becomes hidden in `solomon_proxy.cpp`, the problem will still be present in the config class name.